src/plugins/vis_types/timeseries/public/convert_to_lens/lib/metrics/filter_ratio_formula.ts

```diff
@@ -7,50 +7,59 @@
  */
 
 import type { Query } from '@kbn/es-query';
+import type { Metric, MetricType } from '../../../../common/types';
+import { getFormulaFromMetric, SupportedMetric, SUPPORTED_METRICS } from './supported_metrics';
 import { addTimeRangeToFormula } from '.';
-import type { Metric } from '../../../../common/types';
-import { SUPPORTED_METRICS } from './supported_metrics';
 
 const escapeQuotes = (str: string) => {
   return str?.replace(/'/g, "\\'");
 };
 
-const constructFilterRationFormula = (operation: string, metric?: Query, window?: string) => {
+const constructFilterRationFormula = (
+  operation: string,
+  metric?: Query,
+  reducedTimeRange?: string
+) => {
   return `${operation}${metric?.language === 'lucene' ? 'lucene' : 'kql'}='${
     metric?.query && typeof metric?.query === 'string'
       ? escapeQuotes(metric?.query)
       : metric?.query ?? '*'
-  }'${addTimeRangeToFormula(window)})`;
+  }'${addTimeRangeToFormula(reducedTimeRange)})`;
 };
 
-export const getFilterRatioFormula = (currentMetric: Metric, window?: string) => {
+export const getFilterRatioFormula = (currentMetric: Metric, reducedTimeRange?: string) => {
   // eslint-disable-next-line @typescript-eslint/naming-convention
   const { numerator, denominator, metric_agg, field } = currentMetric;
-  let aggregation = SUPPORTED_METRICS.count;
+  let aggregation: SupportedMetric | null | undefined = SUPPORTED_METRICS.count;
   if (metric_agg) {
-    aggregation = SUPPORTED_METRICS[metric_agg];
+    aggregation = SUPPORTED_METRICS[metric_agg as MetricType];
     if (!aggregation) {
       return null;
     }
   }
-  const operation =
-    metric_agg && metric_agg !== 'count' ? `${aggregation.name}('${field}',` : 'count(';
+  const aggFormula = getFormulaFromMetric(aggregation);
+
+  const operation = metric_agg && metric_agg !== 'count' ? `${aggFormula}('${field}',` : 'count(';
 
   if (aggregation.name === 'counter_rate') {
-    const numeratorFormula = constructFilterRationFormula(
-      `${aggregation.name}(max('${field}',`,
+    const numeratorFormula = `max(${constructFilterRationFormula(
```

Comment:
some left-over max differences which should be counter_rate

Reply:
done